src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -147,94 +157,77 @@ public NetworkPeerServer(Network network, ProtocolVersion version, int internalP
         /// Starts listening on the server's initialialized endpoint.
         /// </summary>
         /// <param name="maxIncoming">Maximal number of newly connected clients waiting to be accepted.</param>
-        public void Listen(int maxIncoming = 8)
+        public void Listen(int maxIncoming = 20)
```

Comment:
good question - i actually don't know, just 8 seemed too low
this is a setting that tells you how many new connections can be queued, so basically if your server is not in trouble, you do not expect the queue to be longer than 1 because accepting new connection is super fast, but if you are doing some calculations, CPU is on 100 %, queue may happen, then if you have 8 pending connections, new clients would be told that they were not accepted. if we increase the number, we have some very little memory cost for that and the clients just wait and there is a good chance they will be accepted 
on high performance servers, this number is set to hundreds or more, but we are not looking for such a high performance, so we do not need 100 or 1000 here. 

but it does no harm to have 100 here, the memory cost is quite low, so therefore little higher value than needed is probably better than little lower and 8 seemed little lower, although in 99.9 % cases it would be well enough ...

Reply:
Nice, thanks.